core/relay/adaptor/text-embeddings-inference/adaptor.go

```diff
@@ -0,0 +1,140 @@
+package textembeddingsinference
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/bytedance/sonic/ast"
+	"github.com/gin-gonic/gin"
+	"github.com/labring/aiproxy/core/common"
+	"github.com/labring/aiproxy/core/model"
+	"github.com/labring/aiproxy/core/relay/adaptor/openai"
+	"github.com/labring/aiproxy/core/relay/meta"
+	"github.com/labring/aiproxy/core/relay/mode"
+	relaymodel "github.com/labring/aiproxy/core/relay/model"
+	"github.com/labring/aiproxy/core/relay/utils"
+)
+
+// text-embeddings-inference adaptor supports rerank and embeddings models deployed by https://github.com/huggingface/text-embeddings-inference
+type Adaptor struct {
+	openai.Adaptor
+}
+
+// base url for text-embeddings-inference, fake
+const baseURL = "https://api.text-embeddings.net/v1"
+
+func (a *Adaptor) GetBaseURL() string {
+	return baseURL
+}
+
+func (a *Adaptor) GetModelList() []*model.ModelConfig {
+	return ModelList
+}
+
+func (a *Adaptor) GetRequestURL(meta *meta.Meta) (string, error) {
+	switch meta.Mode {
+	case mode.Rerank:
+		return a.GetBaseURL() + "/rerank", nil
+	case mode.Embeddings:
+		return a.GetBaseURL() + "/embeddings", nil
+	default:
+		return "", fmt.Errorf("unsupported mode: %s", meta.Mode)
+	}
+}
+
+// text-embeddings-inference api see https://huggingface.github.io/text-embeddings-inference/#/Text%20Embeddings%20Inference/rerank
+
+func (a *Adaptor) SetupRequestHeader(meta *meta.Meta, _ *gin.Context, req *http.Request) error {
+	switch meta.Mode {
+	case mode.Rerank:
+		req.Header.Set("Content-Type", "application/json")
+	case mode.Embeddings:
+		req.Header.Set("Content-Type", "application/json")
+	default:
+		return fmt.Errorf("unsupported mode: %s", meta.Mode)
+	}
+	return nil
+}
+
+func (a *Adaptor) ConvertRequest(meta *meta.Meta, req *http.Request) (string, http.Header, io.Reader, error) {
+	switch meta.Mode {
+	case mode.Rerank:
+		// Parse request body into AST node
+		node, err := common.UnmarshalBody2Node(req)
+		if err != nil {
+			return "", nil, nil, fmt.Errorf("failed to parse request body: %w", err)
+		}
+
+		// Get the documents array and rename it to texts
+		documentsNode := node.Get("documents")
+		if !documentsNode.Exists() {
+			return "", nil, nil, errors.New("documents field not found")
+		}
+
+		// Get the raw documents value
+		documentsValue, err := documentsNode.Raw()
+		if err != nil {
+			return "", nil, nil, fmt.Errorf("failed to get documents value: %w", err)
+		}
+
+		// Set the texts field with the documents value
+		_, err = node.Set("texts", ast.NewString(documentsValue))
+		if err != nil {
+			return "", nil, nil, fmt.Errorf("failed to set texts field: %w", err)
+		}
```

Comment:
do not Set `documentsNode.Raw()`, which is a string. It will become `"texts": "...."` after MarshalJSON.